__tests__/Unit/Components/Tasks/TaskDetails.test.tsx

```diff
@@ -191,228 +190,211 @@ describe('TaskDetails Page', () => {
-        });
-        expect(screen.queryByText(/Successfully saved/i)).toBeNull();
     });
 
-    test('Should render No task progress', async () => {
-        renderWithRouter(
-            <Provider store={store()}>
-                <TaskDetails taskID={details.taskID} />
-            </Provider>
-        );
-        let progressUpdatesSection;
-        await waitFor(() => {
-            progressUpdatesSection = screen.getByText('Progress Updates');
-        });
-        expect(progressUpdatesSection).toBeInTheDocument();
-        const noProgressText = screen.getByText('No Progress found');
-        expect(noProgressText).toBeInTheDocument();
+    await waitFor(() => {
+        const saveButton = screen.getByRole('button', { name: 'Save' });
+        fireEvent.click(saveButton);
+        const editButton = screen.getByRole('button', { name: 'Edit' });
+        expect(editButton).toBeInTheDocument();
     });
+});
 
-    test('should call progress details query', async () => {
-        const spyfn = jest.spyOn(progressQueries, 'useGetProgressDetailsQuery');
-        renderWithRouter(
-            <Provider store={store()}>
-                <TaskDetails taskID={details.taskID} />
-            </Provider>
-        );
-        let progressUpdatesSection;
-        await waitFor(() => {
-            progressUpdatesSection = screen.getByText('Progress Updates');
-        });
-        expect(progressUpdatesSection).toBeInTheDocument();
-        expect(spyfn).toBeCalled();
+test('should update the title and description with the new values', async () => {
+    server.use(...taskDetailsHandler);
+
+    renderWithRouter(
+        <Provider store={store()}>
+            <TaskDetails taskID={details.taskID} />
+            <ToastContainer />
+        </Provider>,
+        {}
+    );
+    await waitFor(() => {
+        const editButton = screen.getByRole('button', { name: 'Edit' });
+        fireEvent.click(editButton);
     });
+    const textareaElement = screen.getByTestId('title-textarea');
+    fireEvent.change(textareaElement, {
+        target: { name: 'title', value: 'New Title' },
+    });
+    await waitFor(async () => {
+        const saveButton = screen.getByRole('button', { name: 'Save' });
+        fireEvent.click(saveButton);
+        expect(await screen.findByText(/Successfully saved/i)).not.toBeNull();
+    });
+});
+test('should not update the title and description with the same values', async () => {
+    server.use(...taskDetailsHandler);
+    renderWithRouter(
+        <Provider store={store()}>
+            <TaskDetails taskID={details.taskID} />
+            <ToastContainer />
+        </Provider>,
+        {}
+    );
+    const editButton = await screen.findByRole('button', { name: 'Edit' });
+    fireEvent.click(editButton);
+    const textareaElement = await screen.findByTestId('title-textarea');
+    fireEvent.change(textareaElement, {
+        target: { name: 'title', value: 'test 1 for drag and drop' },
```

Comment:
Q. the text area already contains the text `test 1 for drag and drop`, right? That's why it will not update

Reply:
Thank you for your feedback and suggestions! You're correct. Since the textarea initially contains the text `test 1 for drag and drop,` attempting to update it with the same value may not trigger a change event.

I've modified the test code to simulate a change event by temporarily setting the value to something different before changing it back to the original value. This ensures that the change event is properly triggered, allowing us to test the behaviour of not updating when the same value is provided.

Thank you for pointing out this potential issue.